lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1167,6 +1292,32 @@ class TextEditingChannel {
     window._replyToPlatformMessage(callback, codec.encodeSuccessEnvelope(true));
   }
 
+  /// Used for submitting the forms attached on the DOM.
+  ///
+  /// Browser will save the information entered to the form.
+  ///
+  /// Called when the form is finalized with save option `true`.
+  /// See: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/text_input.dart#L1277
+  void saveForms() {
+    formsOnTheDom.forEach((String identifier, html.FormElement form) {
+      final html.InputElement submitBtn =
+          form.getElementsByClassName('submitBtn').first as html.InputElement;
+      submitBtn.click();
+    });
+  }
+
+  /// Used for removing the forms on the DOM.
+  ///
+  /// Called when the form is finalized.
+  void cleanForms() {
+    while (formsOnTheDom.length > 0) {
+      final html.FormElement form =
+          formsOnTheDom[formsOnTheDom.entries.last.key] as html.FormElement;
+      form.remove();
+      formsOnTheDom.remove(formsOnTheDom.entries.last.key);
+    }
```

Comment:
To avoid the cast, you can try something like:
```dart
for (html.FormElement form in formsOnTheDom.values) {
  form.remove();
}
formsOnTheDom.clear();
```

Reply:
It will give a concurrent modification exception though, right?

Reply:
We are not modifying the map inside the loop. Inside the loop we are only removing the elements from the DOM. After the loop is done, we clear the map.

Reply:
thanks!